compiler/jsgen.nim

```diff
@@ -897,7 +896,7 @@ proc genCaseJS(p: PProc, n: PNode, r: var TCompRes) =
   of tyString:
     useMagic(p, "toJSStr")
     lineF(p, "switch (toJSStr($1)) {$n", [cond.rdLoc])
-  of tyFloat..tyFloat128:
+  of tyFloat..tyFloat128, tyInt..tyInt64, tyUInt..tyUInt64:
```

Comment:
Does it mean switch case downgrades to if for int types? Will it introduce performance regression?

Reply:
only numberic will downgrades to `if`, I dont think it will introduce performance regression, it will avoid huge iteration at compile time, and reduce code gen time.